Modules/Core/Common/include/itkImageAlgorithm.hxx

```diff
@@ -197,24 +197,28 @@ ImageAlgorithm::EnlargeRegionOverBox(const typename InputImageType::RegionType &
 {
   typename OutputImageType::RegionType outputRegion;
 
-  // Get the index of the corners of the input region,
-  // map them to physical space, and convert them
-  // to index for this image.
-  // The region has 2^ImageDimension corners, each
-  // of them either on the inferior or superior edge
+  // Get the index of the corners of the input region, map them to input physical space.
+  // Convert input physical corners to output physical corners using transform
+  // and finally map output physical corners to index of output image.
+  // The input region has 2^ImageDimension corners, each
+  // of which is either on the inferior or superior edge
   // along each dimension.
-  unsigned int numberOfCorners = 1;
+  unsigned int numberOfInputCorners = 1;
   for (unsigned int dim = 0; dim < OutputImageType::ImageDimension; ++dim)
   {
-    numberOfCorners *= 2;
+    numberOfInputCorners *= 2;
   }
-  using ContinuousIndexType = ContinuousIndex<double, OutputImageType::ImageDimension>;
-  std::vector<ContinuousIndexType> corners(numberOfCorners);
+  using ContinuousInputIndexType = ContinuousIndex<double, InputImageType::ImageDimension>;
+  using ContinuousOutputIndexType = ContinuousIndex<double, OutputImageType::ImageDimension>;
 
-  for (unsigned int count = 0; count < numberOfCorners; ++count)
+  std::vector<ContinuousInputIndexType>  inputCorners(numberOfInputCorners);
```

Comment:
If I understand it correctly, we are trying to map each input corner to output space. If the dimensions decrease, two corners could map to a single corner or a different point inside the output region. So 'outputCorners'   could contain points which are not 'corners', but that shouldn't be an issue.
As and example consider a parallelepiped in 3D, and a transform which project it to say x-y plane. Some corner of   the parallelepiped can map to inside the output 2D parrellogram, so not all 'outputCorners' will be corners, but that shouldn't be an issue since following code which computes output region calculates maximum/minimum index over all points. 
 Of course this function won't work for non-linear transformations which map non-corners to corners(Not sure if there is a name for such transforms?).

Reply:
OK, this makes sense.